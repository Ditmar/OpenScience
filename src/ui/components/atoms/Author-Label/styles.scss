@import '../../../../globals/fonts';
@import '../../../../globals/mixins';
@import '../../../../globals/variables';


.author{
  color: $ads-text-main-hover;
  font-family: Lato, sans-serif;
  font-size: 0.625rem;
  font-weight: 275;
  line-height: 0.85rem;
   
  @include respond-to('small') {
    font-size: 0.625rem;
  }

  @include respond-to('medium') {
    font-size: 1rem;
  }

  @include respond-to('large') {
    font-size: 1.5rem;
  }
}


