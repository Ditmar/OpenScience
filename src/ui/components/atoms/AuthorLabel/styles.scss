@import '../../../../globals/breakpoints';
@import '../../../../globals/mixins';
@import '../../../../globals/variables';

.author-label{
  font-size: $ads-size-main;
  margin-left: 1rem;
  margin-right: 1rem;
  color: $ads-text-main;
  font-family: Arial, Helvetica, sans-serif;
  font-size: $ads-size-main;
  padding: 1.26rem;
  width: 100%;
  border: 2px solid transparent;

  @include respond-to('medium') {
    width: 80%;
  }

  @include respond-to('large') {
    width: 30%;
  }

  @include respond-to('extra-large') {
    width: 30%;
  }
}
