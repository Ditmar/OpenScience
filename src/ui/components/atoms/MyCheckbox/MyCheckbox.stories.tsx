import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import { ModeSwitcher } from '../../../../style-library/core/ThemeSwitcher';
import MyCheckbox from './MyCheckbox';

const meta: Meta<typeof MyCheckbox> = {
  title: 'Atoms/MyCheckbox',
  component: MyCheckbox,
  argTypes: {
    checked: {
      control: 'boolean',
      description: 'If true, the component is checked.',
      defaultValue: false,
    },
    onChange: {
      description: 'Callback fired when the state of the checkbox changes.',
    },
    disabled: {
      control: 'boolean',
      description: 'If true, the component is disabled.',
      defaultValue: false,
    },
    variant: {
      control: 'select',
      options: ['primary', 'secondary', 'info', 'danger', 'success', 'warning', 'default'],
      description: 'The color variant of the checkbox.',
      defaultValue: 'default',
    },
  },
  decorators: [
    (Story) => (
      <div style={{ display: 'flex', flexDirection: 'column', gap: '20px', padding: '20px' }}>
        <ModeSwitcher />
        <hr style={{ width: '100%', borderTop: '1px solid #eee', margin: '20px 0' }} />
        <Story />
      </div>
    ),
  ],
  args: {
    onChange: () => {},
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Unchecked: Story = {
  args: {
    // checked se omite para false (es el valor por defecto)
    // onChange ya lo tiene el args global en meta
  },
};

export const Checked: Story = {
  args: {
    checked: true,
  },
};

export const DisabledUnchecked: Story = {
  args: {
    disabled: true,
  },
};

export const DisabledChecked: Story = {
  args: {
    checked: true,
    disabled: true,
  },
};

export const Variants: Story = {
  render: (args) => (
    <div style={{ display: 'flex', flexDirection: 'column', gap: '10px' }}>
      <MyCheckbox checked={args.checked} onChange={args.onChange} variant="info" />
      <MyCheckbox checked={args.checked} onChange={args.onChange} variant="danger" />
      <MyCheckbox checked={args.checked} onChange={args.onChange} variant="success" />
      <MyCheckbox checked={args.checked} onChange={args.onChange} variant="warning" />
      <MyCheckbox checked={args.checked} onChange={args.onChange} variant="default" />
    </div>
  ),
  args: {
    checked: true,
  },
};

export const UncheckedWithValue: Story = {
  args: {
    value: 'my-checkbox-value',
  },
};

export const CheckedWithValue: Story = {
  args: {
    checked: true,
    value: 'another-checkbox-value',
  },
};
