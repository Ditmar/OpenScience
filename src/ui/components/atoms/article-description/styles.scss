@import '../../../../globals/variables';
@import '../../../../globals/mixins';

.article-description__text {
  font-family: Lato, sans-serif;
  font-size: $ads-size-main;
  font-weight: 275;
  letter-spacing: 0.125rem;
  line-height: 2.05rem;
  margin-bottom: $ads-letter-spacing;

  @font-face {
    font-display: swap;
    font-family: Lato;
    font-style: normal;
    src: url('/fonts/Lato-Regular.woff2') format('woff2');
  }

  @include respond-to('small') {
    font-size: 1rem;
    letter-spacing: 0.075rem;
    margin-bottom: 0.75rem;
  }

  @include respond-to('medium') {
    font-size: 1.25rem;
    letter-spacing: 0.1rem;
    margin-bottom: 1rem;
  }

  @include respond-to('large') {
    font-size: 1.5rem;
    letter-spacing: 0.125rem;
    margin-bottom: 1.25rem;
  }

  @include respond-to('extra-large') {
    font-size: 1.5rem;
    letter-spacing: 0.15rem;
  }
}

.bold-text {
  font-weight: $ads-font-weight-bold;
}

.italic-text {
  font-style: italic;
}

.underline-text {
  text-decoration: underline;
}

.highlight-text {
  background-color: $ads-primary-main;
}
