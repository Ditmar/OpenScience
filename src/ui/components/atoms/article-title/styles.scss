@import '../../../../globals/variables';
@import '../../../../globals/mixins';

.title-text {
    font-family: Roboto, sans-serif;
    font-size: 2.5rem;
    font-weight: 800;
    letter-spacing: 0.001rem;
    line-height: 3.492rem;

    display: flex;

  
    @include respond-to ('small'){
        font-size: 1rem;
        letter-spacing: 0.075rem;
    }
  
    @include respond-to ('medium'){
        font-size: 1.25rem;
        letter-spacing: 0.1rem;
    }
  
    @include respond-to ('large'){
        font-size: 1.5rem;
        letter-spacing: 0.01rem;
    }
  
    @include respond-to ('extra-large'){
        font-size: 1.5rem;
        letter-spacing: 0.01rem;
    }
  
}
  
.bold-text {
    font-weight: $ads-font-weight-bold;
}
