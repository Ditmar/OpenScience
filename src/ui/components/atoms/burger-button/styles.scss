@import '../../../../globals/variables';
@import '../../../../globals/mixins';

.burger-button {
    background: none;
    border: none;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 24px;
    width: 30px;
    padding: 0;
  }
  
  .burger-line {
    width: 100%;
    height: 3px;
    background-color: #333;
    transition: all 0.3s ease;
  }
  
  .burger-button.active .burger-line:nth-child(1) {
    transform: rotate(45deg) translate(5px, 5px);
  }
  
  .burger-button.active .burger-line:nth-child(2) {
    opacity: 0;
  }
  
  .burger-button.active .burger-line:nth-child(3) {
    transform: rotate(-45deg) translate(5px, -5px);
  }
  
