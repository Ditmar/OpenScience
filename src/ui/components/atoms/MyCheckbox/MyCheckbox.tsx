// src/components/atoms/MyCheckbox/MyCheckbox.tsx
import React from 'react';
import type { MyCheckboxProps } from './Types/IProps';
import { StyledCheckbox, muiColorMap } from './Mycheckbox.styles';

function MyCheckbox({
  checked = false,
  onChange,
  disabled = false,
  variant = 'default',

  value,
  id,
  name,
  required,
  readOnly,
  tabIndex,
  className,
  style,
  onBlur,
  onFocus,
}: MyCheckboxProps) {
  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.checked);
  };

  const muiColor = muiColorMap[variant] ?? 'default';

  return (
    <StyledCheckbox
      checked={checked}
      onChange={handleCheckboxChange}
      disabled={disabled}
      color={muiColor}
      inputProps={{
        'aria-checked': checked,
        'aria-disabled': disabled,
        role: 'checkbox',
      }}
      customVariant={variant}
      value={value}
      id={id}
      name={name}
      required={required}
      readOnly={readOnly}
      tabIndex={tabIndex}
      className={className}
      style={style}
      onBlur={onBlur}
      onFocus={onFocus}
    />
  );
}

export default MyCheckbox;
