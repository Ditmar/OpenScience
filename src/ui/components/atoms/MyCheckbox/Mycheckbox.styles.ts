import { styled } from '@mui/material/styles';
import Checkbox, { type CheckboxProps } from '@mui/material/Checkbox';
import type { MyCheckboxVariant } from './Types/IProps';

export const muiColorMap: Record<MyCheckboxVariant, CheckboxProps['color']> = {
  default: 'default',
  success: 'success',
  info: 'info',
  warning: 'warning',
  danger: 'error',
};

export const StyledCheckbox = styled(Checkbox)<{ customVariant: MyCheckboxVariant }>(
  ({ theme }) => ({
    borderRadius: theme.shape.borderRadius || 4,
  }),
);
