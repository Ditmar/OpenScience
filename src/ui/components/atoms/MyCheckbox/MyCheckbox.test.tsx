// src/components/atoms/MyCheckbox/MyCheckbox.test.tsx

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { vi } from 'vitest';
import { ThemeProvider } from '@mui/material/styles';
import { lightTheme } from '../../../../style-library/themes/default';

import MyCheckbox from './MyCheckbox';
import '@testing-library/jest-dom';

describe('MyCheckbox (Without test-utils.tsx)', () => {
  const mockOnChange = vi.fn();

  beforeEach(() => {
    mockOnChange.mockClear();
  });

  const renderWithTheme = (component: React.ReactElement) => {
    return render(<ThemeProvider theme={lightTheme}>{component}</ThemeProvider>);
  };

  test('should render and be unchecked by default', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
  });

  test('should render and be checked when `checked` prop is true', () => {
    renderWithTheme(<MyCheckbox checked onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeChecked();
  });

  test('should render and be disabled when `disabled` prop is true', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} disabled />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeDisabled();
  });

  test('should not be disabled by default', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).not.toBeDisabled();
  });

  test('should call `onChange` with true when clicked and unchecked', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);
    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith(true);
  });

  test('should call `onChange` with false when clicked and checked', () => {
    renderWithTheme(<MyCheckbox checked onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);
    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith(false);
  });

  test('should NOT call `onChange` when disabled and clicked', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} disabled />);
    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);
    expect(mockOnChange).not.toHaveBeenCalled();
  });

  test('should have the correct `role="checkbox"`', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toHaveAttribute('role', 'checkbox');
  });

  test('should have the correct `aria-checked` attribute', () => {
    const { rerender } = renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toHaveAttribute('aria-checked', 'false');

    rerender(
      <ThemeProvider theme={lightTheme}>
        <MyCheckbox checked onChange={mockOnChange} />
      </ThemeProvider>,
    );
    expect(checkbox).toHaveAttribute('aria-checked', 'true');
  });

  test('should have the `aria-disabled` attribute when disabled', () => {
    renderWithTheme(<MyCheckbox onChange={mockOnChange} disabled />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toHaveAttribute('aria-disabled', 'true');
  });

  test('should match snapshot when unchecked by default', () => {
    const { asFragment } = renderWithTheme(<MyCheckbox onChange={mockOnChange} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
