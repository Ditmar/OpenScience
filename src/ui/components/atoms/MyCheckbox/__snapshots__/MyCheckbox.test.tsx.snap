// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyCheckbox (Without test-utils.tsx) > should match snapshot when unchecked by default 1`] = `
<DocumentFragment>
  <span
    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorDefault MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorDefault MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorDefault MuiCheckbox-sizeMedium css-doy2y2-MuiButtonBase-root-MuiCheckbox-root"
    customvariant="default"
  >
    <input
      aria-checked="false"
      aria-disabled="false"
      class="PrivateSwitchBase-input css-j8yymo"
      data-indeterminate="false"
      role="checkbox"
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-eqnrfk-MuiSvgIcon-root"
      data-testid="CheckBoxOutlineBlankIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
